src/ansys/aedt/core/modeler/circuits/primitives_circuit.py

```diff
@@ -1239,6 +1239,23 @@ def create_wire(self, points, name=""):
         except Exception:
             return False
 
+    @pyaedt_function_handler()
+    def component_path(self, component_name):
+        """Component definition path.
+        Parameters
+        ----------
+        component_name: str
+            Name of the component.
+        """
+        for component in self.components.values():
+            if component.component_info["InstanceName"] == component_name:
+                component_definition = component.component_info["Info"]
+                if not self.o_component_manager.GetData(component_definition):
+                    self.logger.warning("Component has no path")
+                    return False
+                else:
+                    return (self.o_component_manager.GetData(component_definition)[24][1][12][1].split(" ")[1])[1:-1]
```

Comment:
Hi @gkorompi , THis is nice, maybe it would be better to have the component path at component level, so I would do it as a property like>

component.component_path 

What do you think?

If you like it, I think it should be at CircuitComponent level.

Reply:
@Samuelopez-ansys I agree! That was my initial thought, but when in CircuitComponent level the function o_component_manager is not there (it's in the level before that) that is why i implemented it there. Let me know what do you think. Thanks!

Reply:
If I am not wrong you can access: self._circuit_components.o_component_manager